docs/teachertool/validator-plans-shared.json

```diff
@@ -70,6 +70,32 @@
                     }
                 }
             ]
+        },
+        {
+            ".desc": "A variable's value is set",
+            "name": "variable_set",
+            "threshold": 1,
+            "checks": [
+                {
+                    "validator": "blocksExist",
+                    "blockCounts": {
+                        "variables_set": 1
```

Comment:
We may want to add an additional check here for variables_change, which will also work as a set in some cases (but keep threshold as 1).

Reply:
I would prefer to keep that as a separate check and wait for feedback on it. It makes sense that we assign a variable a value when the variable is created so there are less errors. However, allowing a `variables_change`  to act as a setter in a validation scenario feels like a dangerous thing to allow. The act of distinctively setting a variable value is a pretty important learning point for computer science.

Reply:
I can add a validator for `variables_change`, too, though.

Reply:
Hm, I don't think we need to add a whole separate validator for it. Feel free to leave as-is for now.

Reply:
Validating that the program changes the value of a variable is actually non-trivial, since both "change" or "set" could do it, but "set" would have to be called twice on the same variable to achieve the intuitive result, which might be tricky to check. I'd only go down that rabbit hole if we get feedback asking for it.

Reply:
Sounds good. I'll leave as is, then. Will definitely want to revisit the "variable changed" validation in the future.